torch/distributed/_tensor/_dispatch.py

```diff
@@ -231,6 +231,11 @@ def default_tensor(spec: DTensorSpec) -> torch.Tensor:
             if output_sharding.output_spec is not None:
                 return args[0]
             else:
+                # TODO: this would only work for 1-d mesh
```

Comment:
If this only works for 1D mesh, I would recommend checking and raising an error for 2D+ with a helpful error message rather than waiting for a downstream error.

Reply:
Switch to DTensor, calling `reduce` and `full_tensor`, back to tensor in the current commit
Could work without limitation of 1-d mesh now.